dask/array/tests/test_array_core.py

```diff
@@ -1209,6 +1210,24 @@ def test_reshape_unknown_dimensions():
     pytest.raises(ValueError, lambda: da.reshape(a, (-1, -1)))
 
 
+@pytest.mark.parametrize(
+    "shape, chunks, reshape_size",
+    [
+        # Test reshape where output chunks would otherwise be too large
+        ((300, 180, 4, 18483), (-1, -1, 1, 183), (300, 180, -1)),
+        # Test reshape where multiple chunks match between input and output
+        ((300, 300, 4, 18483), (-1, -1, 1, 183), (300, 300, -1)),
+    ],
+)
+def test_reshape_avoids_large_chunks(shape, chunks, reshape_size):
+    limit = parse_bytes("128MiB")
```

Comment:
Sorry, just getting around to looking at this now. IIUC it looks like this `limit` isn't being passed down to `reshape`, so this test is checking the default limit, but not a user-specified one. Looking a bit more, we should probably add a `limit=` keyword to `Array.reshape` as well as the top-level `reshape` function

Reply:
Also, maybe this line should use `config.get("array.chunk-size")` in case the default value of 128 MiB changes in the future

Reply:
These are all good suggestions, I've opened https://github.com/dask/dask/pull/8494 to include them